composer/packages/diagram/src/views/components/block.tsx

```diff
@@ -2,16 +2,46 @@
 import { Block as BlockNode } from "@ballerina/ast-model";
 import * as React from "react";
 import { DiagramUtils } from "../../diagram/diagram-utils";
+import { BlockViewState } from "../../view-model/block";
+import { BlockDropdown } from "./block-dropdown";
 
-export const Block: React.StatelessComponent<{
-        model: BlockNode
-    }> = ({
-        model
-    }) => {
+export class Block extends React.Component<{ model: BlockNode }, { isHovered: boolean }> {
+
+    public state = {
+        isHovered: false
+    };
+
+    public render() {
+        const { model }  = this.props;
+        const { isHovered } = this.state;
         const statements: React.ReactNode[] = [];
+        const viewState: BlockViewState = model.viewState;
+        const { x, y, w, h, leftMargin } = viewState.bBox;
+        const triggerPosition = viewState.menuTrigger;
+        const hoverRectBBox = {
+            height: h, width : w + leftMargin, x: x - leftMargin,  y
+        };
         statements.push(DiagramUtils.getComponents(model.statements));
         return (
-            <g className="panel">
+            <g
+                className="panel block"
+            >
                 {statements}
```

Comment:
`statements` is a blank array?

Reply:
nope, see statements.push(DiagramUtils.getComponents(model.statements));